### PR TITLE
Update to tensorflowjs 1.0.1

### PR DESCRIPTION
**Pull request by javyxx**
Requiere a release of tfjs-image-recognition-base first

### CODE REVIEW

**Comment by hanjeahwan** [comment]
💖

**Comment by justadudewhohacks** [comment]
Thanks!

I just published tfjs-image-recognition-base@0.5.0, you can update the package in your PR now. Please remove the changes from build and dist in this PR as well.

**Comment by javyxx** [comment]
Done. I do some additional change , replace deprecated api .
    out = tf.batchNormalization<tf.Rank.R4>(
    out = tf.batchNorm<tf.Rank.R4>(

File: src/mtcnn/stage2.ts [hunk 1]
```diff
@@ -56,10 +56,10 @@ export async function stage2(
...
     const regions = indicesNms.map(idx =>
       new MtcnnBox(
-        rnetOuts[indices[idx]].regions.get(0, 0),
-        rnetOuts[indices[idx]].regions.get(0, 1),
-        rnetOuts[indices[idx]].regions.get(0, 2),
-        rnetOuts[indices[idx]].regions.get(0, 3)
+        rnetOuts[indices[idx]].regions.arraySync()[0][0],
+        rnetOuts[indices[idx]].regions.arraySync()[0][1],
+        rnetOuts[indices[idx]].regions.arraySync()[0][2],
+        rnetOuts[indices[idx]].regions.arraySync()[0][3]
```
Comment (justadudewhohacks) [inline src/mtcnn/stage2.ts]: Better call arraySync once outside of the map callback as you did [here](https://github.com/justadudewhohacks/face-api.js/pull/246/files#diff-cb18491d4c23fc57b660e5a7b2287623R50).

File: src/mtcnn/stage3.ts [hunk 2]
```diff
@@ -40,10 +40,10 @@ export async function stage3(
     .map(({ idx }) => idx)
 
   const filteredRegions = indices.map(idx => new MtcnnBox(
-    onetOuts[idx].regions.get(0, 0),
-    onetOuts[idx].regions.get(0, 1),
-    onetOuts[idx].regions.get(0, 2),
-    onetOuts[idx].regions.get(0, 3)
+    onetOuts[idx].regions.arraySync()[0][0],
```
Comment (justadudewhohacks) [inline src/mtcnn/stage3.ts]: Same here.

File: src/mtcnn/stage3.ts [hunk 3]
```diff
@@ -69,8 +69,8 @@ export async function stage3(
     points = indicesNms.map((idx, i) =>
       Array(5).fill(0).map((_, ptIdx) =>
         new Point(
-          ((onetOuts[idx].points.get(0, ptIdx) * (finalBoxes[i].width + 1)) + finalBoxes[i].left) ,
-          ((onetOuts[idx].points.get(0, ptIdx + 5) * (finalBoxes[i].height + 1)) + finalBoxes[i].top)
+          ((onetOuts[idx].points.arraySync()[0][ptIdx] * (finalBoxes[i].width + 1)) + finalBoxes[i].left) ,
```
Comment (justadudewhohacks) [inline src/mtcnn/stage3.ts]: Same here.

File: src/mtcnn/stage2.ts (hunk 1 above)
Comment (javyxx) [reply src/mtcnn/stage2.ts]: Yep. But is imposible outside de map because var idx .

**Comment by justadudewhohacks** [comment]
Looks fine now, no worries.

**Comment by justadudewhohacks** [comment]
Hmm, I want to hold merging this PR for a bit, since something seems to be fishy with tfjs-core version 1.0.1. Compared to 0.14.2 it seems to consume significantly more GPU memory. This causes WebGL exception when running the unit tests locally with latest chrome on the WebGL backend.

**Comment by javyxx** [comment]
The variables that we have defined may prevent the release of memory.  We can't execute tf.disponse

**Comment by javyxx** [comment]
Or maybe is a issue of tf 1.0.1

**Comment by justadudewhohacks** [comment]
Awaiting respone from https://github.com/tensorflow/tfjs/issues/1440 first.
